File: programs/reusing-classes_refactoring-1.java
public class Automaton {
    private GridCanvas grid;

    public void run(String title, int rate) {
        JFrame frame = new JFrame(title);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setResizable(false);
        frame.add(this.grid);
        frame.pack();
        frame.setVisible(true);
        this.mainloop(rate);
    }
}

File: programs/reusing-classes_refactoring-1.cpp
#include <iostream>
// You will need to include appropriate headers for your GUI framework (e.g.,
// GLFW, SDL, Qt). This example uses placeholders to simulate the GUI elements.
class GridCanvas {};
class Automaton {
private:
  GridCanvas* grid;

public:
  Automaton(GridCanvas* grid): grid(grid) {
  }
  ~Automaton() {
    delete grid;
  }
  void run(const string& title, int rate) {
    // JFrame frame(title); // Placeholder for JFrame creation
    // frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE); // Placeholder
    // frame.setResizable(false); // Placeholder
    // frame.add(this->grid); // Placeholder for adding grid to frame
    // frame.pack(); // Placeholder
    // frame.setVisible(true); // Placeholder
    mainloop(rate);
  }

protected:
  void mainloop(int rate) {
    // Implementation of mainloop - replace with your actual game loop
  }
};